storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -950,20 +1074,33 @@
             TOPOLOGY-EVENTLOGGER-EXECUTORS (total-conf TOPOLOGY-EVENTLOGGER-EXECUTORS)
             TOPOLOGY-MAX-TASK-PARALLELISM (total-conf TOPOLOGY-MAX-TASK-PARALLELISM)})))
 
+(defn blob-rm [blob-store key storm-cluster-state]
+  (try
+    (.deleteBlob blob-store key (get-nimbus-subject))
+    (if (instance? LocalFsBlobStore blob-store)
+      (.remove-blobstore-key! storm-cluster-state key))
+    (catch Exception e)))
+
+(defn rm-from-blob-store [id blob-store storm-cluster-state]
```

Comment:
Could we rename these?  blob-rm and rm-from-blob-store don't differentiate each other.  I think it would be better to be something like `blob-rm-key` and `blob-rm-topology-keys`